npm_and_yarn/helpers/lib/npm6/updater.js

```diff
@@ -138,10 +138,10 @@ function installArgs(
     if (existingVersionRequirement.match(desiredVersion)) {
       return `${depName}@${existingVersionRequirement}`;
     } else {
-      return `${depName}@${existingVersionRequirement.replace(
-        /#.*/,
-        ""
-      )}#${desiredVersion}`;
+      if (existingVersionRequirement.includes("#")) {
+        return `${depName}@${existingVersionRequirement.replace(/#.*/, "")}#${desiredVersion}`;
+      }
+      return `${depName}@${existingVersionRequirement}`;
```

Comment:
Nice detective work on finding this fix! 🙌

Reply:
Thanks. This was a fun side quest.